doc/contribute/bin_blobs.rst

```diff
@@ -0,0 +1,231 @@
+
+Software license
+================
+
+Most binary blobs are distributed under proprietary licenses which vary
+significantly in nature and conditions. It is up to the vendor to specify the
+license as part of the blob submission process. Blob vendors may impose a
+click-through or other EULA-like workflow when users fetch and install blobs.
+
+Hosting
+=======
+
+Blobs must be hosted on the Internet and managed by third-party infrastructure.
+Two potential examples are Git repositories and web servers managed by
+individual hardware vendors.
+
+The Zephyr Project will not host binary blobs in its Git repositories or
+anywhere else.
+
+Fetching blobs
+==============
+
+Blobs will be fetched from official third-party sources via a west extension
+command which users will be able to run to this effect. The extension command
+implementation will be hosted in the main zephyr repository. This is required to
+give the project overall control over the mechanism used to fetch the blobs.
+
+The blobs themselves must be specified in the :file:`module.yml` files included
+in separate Zephyr :ref:`module repositories <modules>` maintained by their
+respective vendors.  This means that in order to submit a binary blob to the
+upstream Zephyr distribution, a module repository must exist first or be created
+as part of the submission process.
+
+Each blob which may be fetched must be individually identified in the
+corresponding :file:`module.yml` file. A specification for a blob must contain:
+
+- An abstract description of the blob itself
+- Version information
+- A reference to vendor-provided documentation
+- The blob’s :ref:`type <bin-blobs-types>`, which must be one of the allowed types
+- A checksum for the blob, which the west extension checks after downloading.
+  This is required for reproducibility and to allow bisecting issues as blobs
+  change using Git and west
+- License text applicable to the blob or a reference to such text, in SPDX
+  format
+
+The west extension used to fetch blobs will accept command line arguments that
+will allow the user to select the module or modules to fetch blobs from, as well
+as obtaining information and metadata on available blobs. This is required for
+user control and a uniform user experience regardless of the blob vendor.
+
+The west extension will only fetch and download the binary blobs themselves. Any
+accompanying code must be present in the corresponding module repository.
+
+Tainting
+========
+
+Inclusion of binary blobs will taint the Zephyr build. The definition of
+tainting originates in the `Linux kernel
+<https://www.kernel.org/doc/html/latest/admin-guide/tainted-kernels.html>`_ and,
+in the context of Zephyr, a tainted image will be one that includes binary blobs
+in it.
+
+Tainting will be communicated to the user in the following manners:
+
+- One or more Kconfig options ``TAINT_BLOBS_*`` will be set to ``y``
+- The Zephyr build system, during its configuration phase, will issue a warning
```

Comment:
Missing period at end of sentence in this latest update.

Reply:
Fixed